src/python/pants/rules/core/strip_source_roots_test.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from unittest.mock import Mock
+
+from pants.engine.fs import create_fs_rules
+from pants.engine.legacy.graph import HydratedTarget
+from pants.engine.rules import RootRule
+from pants.engine.selectors import Params
+from pants.rules.core.strip_source_root import SourceRootStrippedSources, strip_source_root
+from pants.source.source_root import SourceRootConfig
+from pants_test.subsystem.subsystem_util import init_subsystem
+from pants_test.test_base import TestBase
+
+
+class StripSourceRootsTests(TestBase):
+  @classmethod
+  def rules(cls):
+    return super().rules() + [
+      strip_source_root,
+      RootRule(SourceRootConfig),
+      RootRule(HydratedTarget),
+    ] + create_fs_rules()
+
```

Comment:
We’ll want to test that we don’t strip Files. I think in your mock function, allow setting `adaptor.type_alias` to whatever desired. Then you can add a new test similar to what you have.

Reply:
These tests have a lot of duplication. Consider converting `mock_hydrated_target` into a function `assert_stripped_source_file(*, original_source_file, expected_source_file, target_type_alias=None)`